app/src/main/java/com/codesoom/assignment/controllers/TaskRepository.java

```diff
@@ -0,0 +1,77 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.models.Task;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * 할 일을 관리하는 클래스이다.
+ */
+public class TaskRepository {
+
+    private static Map<Long, Task> tasks = new LinkedHashMap<>();
+    private static Long taskId = 0L;
+
+    /**
+     * 할 일 목록을 반환한다.
+     * @return 할 일 목록
+     */
+    public ArrayList<Task> findList() {
+        return new ArrayList<>(tasks.values());
+    }
+
+    /**
+     * 할 일을 추가한다.
+     * @param task 추가할 할 일
+     * @return 추가된 할 일
+     */
+    public Task add(Task task) {
+        task.setId(generateId());
+        tasks.put(taskId, task);
+        return task;
+    }
+
+    /**
+     * 할 일을 찾아서 반환한다.
+     * @param id 할 일의 pk
+     * @return 찾은 할 일
+     */
+    public Optional<Task> findDetail(Long id) {
+        Task task = tasks.get(id);
+        return Optional.ofNullable(task);
+    }
+
+    /**
+     * 할 일을 수정한다.
+     * @param id 수정할 할 일의 pk
+     * @param title 수정할 할 일의 제목
+     * @return 수정된 할 일
+     */
+    public Optional<Task> update(Long id, String title) {
```

Comment:
JavaDoc에서는 리턴값에 대한 설명을 분명히 명시해주는 것이 좋습니다. 코드에 다 보이는데 괜찮지 않을까? 하는 생각이 들 수 있는데요, JavaDoc을 빌드하면 빌드된 결과에서는 코드가 사라지고 주석만 남아 문서가 되거든요.

```suggestion
    /**
     * 할 일을 수정하고, 수정된 할 일을 반환합니다.
     *
     * @param id 수정할 할 일의 식별자
     * @param title 수정할 할 일의 제목
     * @return 수정된 할 일
     */
    public Optional<Task> update(Long id, String title) {
```

Reply:
JavaDoc에서 리턴값에 대한 설명을 신경쓰겠습니다.